### PR TITLE
feat(DEW): DEW support a new resource to manage CSMS event

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

DEW support a new resource to manage CSMS event

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- The API response status code is `404` when the resource is not exist.
image

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dew' TESTARGS='-run TestAccCsmsEvent_basic'
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccCsmsEvent_basic -timeout 360m -parallel 4
=== RUN   TestAccCsmsEvent_basic
=== PAUSE TestAccCsmsEvent_basic
=== CONT  TestAccCsmsEvent_basic
--- PASS: TestAccCsmsEvent_basic (27.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       27.397s
```

### CODE REVIEW

File: docs/resources/csms_event.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  Changing this parameter will create a new resource.
+
+* `event_types` - (Required, List) Specifies the event type list. Valid values are:
+  + **SECRET_VERSION_CREATED**: Triggered when a version of a secret is created.
+  + **SECRET_VERSION_EXPIRED**: Triggered when a secret version expires. (Triggered only once for each expiry).
+  + **SECRET_ROTATED**: Triggered when a secret is rotated. Currently, only RDS secrets can be automatically rotated.
+  + **SECRET_DELETED**: Triggered when a secret is deleted.
+
+* `status` - (Required, String) Specifies the event status. Valid values are **ENABLED** and **DISABLED**.
+  Only the event in **ENABLED** status can trigger the event types.
```
Comment (Jason-Zhang9309) [inline docs/resources/csms_event.md]:
```suggestion
  Only the event in **ENABLED** status can be triggered.
```

File: docs/resources/csms_event.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the event name. The valid length is limited from `1` to `64`.
+  Only letters, digits, underscores (_) and hyphens (-) are allowed.
+
+  Changing this parameter will create a new resource.
+
+* `event_types` - (Required, List) Specifies the event type list. Valid values are:
+  + **SECRET_VERSION_CREATED**: Triggered when a version of a secret is created.
+  + **SECRET_VERSION_EXPIRED**: Triggered when a secret version expires. (Triggered only once for each expiry).
```
Comment (Jason-Zhang9309) [inline docs/resources/csms_event.md]: the style of the descriptions should be same

File: huaweicloud/services/dew/resource_huaweicloud_csms_event.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+		product = "kms"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating KMS client: %s", err)
+	}
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{event_name}", d.Get("name").(string))
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_csms_event.go]:
```suggestion
	getPath = strings.ReplaceAll(getPath, "{event_name}", d.Id()))
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_event.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+		return diag.Errorf("error deleting CSMS event: %s", err)
+	}
+
+	return nil
+}
+
+func resourceCsmsEventImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	return []*schema.ResourceData{d}, d.Set("name", d.Id())
+}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_csms_event.go]: this is unnecessary

File: docs/resources/csms_event.md (hunk 1 above)
Comment (deer-hang) [reply docs/resources/csms_event.md]: Done

File: docs/resources/csms_event.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/csms_event.md]: OK

File: huaweicloud/services/dew/resource_huaweicloud_csms_event.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/dew/resource_huaweicloud_csms_event.go]: OK

File: huaweicloud/services/dew/resource_huaweicloud_csms_event.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/dew/resource_huaweicloud_csms_event.go]: OK

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dew' TESTARGS='-run TestAccCsmsEvent_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccCsmsEvent_basic -timeout 360m -parallel 4
=== RUN   TestAccCsmsEvent_basic
=== PAUSE TestAccCsmsEvent_basic
=== CONT  TestAccCsmsEvent_basic
--- PASS: TestAccCsmsEvent_basic (28.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       28.689s
```

**Pull request merged**